### PR TITLE
Adding support for line_items in payment request

### PR DESCRIPTION
**Pull request by aleksen**

### CODE REVIEW

**Comment by aleksen** [comment]
https://github.com/mcash/mcash.loyalty/pull/81 cannot be merged before this has been merged

File: mcash/mapi_client/mapi_client.py [hunk 1]
```diff
@@ -325,6 +325,11 @@ def create_payment_request(self, customer, currency, amount, allow_credit,
                 [{"uri": "http://example.com/uri1",
                   "caption": "This is always shown",
                   "show_on": ["pending", "fail", "ok"]}]
+            line_items:
```
Comment (mandus) [inline mcash/mapi_client/mapi_client.py]: Not sure who this documentation is meant for, but maybe provide a sample, just to make usage clear?

File: mcash/mapi_client/mapi_client.py [hunk 2]
```diff
@@ -404,6 +412,9 @@ def update_payment_request(self, tid, currency=None, amount=None,
                 Action to perform.
             required_scope:
                 Scopes required to fulfill payment
+            line_items:
+                An updated line_items. Will fail if line_items
+                already set in the payment request.
```
Comment (mandus) [inline mcash/mapi_client/mapi_client.py]: or if the total amount of the line items doesn't match the amount in payment. Btw, that goes for line items in general - so probably add that to the doc-string above?

File: mcash/mapi_client/validation.py [hunk 3]
```diff
@@ -71,7 +71,20 @@ def wrapper(*args, **kwargs):
     'text': basestring,
     Required('action'): Any('auth', 'sale', 'AUTH', 'SALE'),
     Required('expires_in'): All(int, Range(min=0, max=2592000)),
-    'links': [{'uri': basestring, 'caption': basestring, 'show_on': [Any('pending', 'fail', 'ok')]}]
+    'links': [{'uri': basestring, 'caption': basestring, 'show_on': [Any('pending', 'fail', 'ok')]}],
+    'line_items': [{
+        Required('product_id'): basestring,
+        Required('vat'): basestring,
```
Comment (mandus) [inline mcash/mapi_client/validation.py]: vat is optional; defaults to

File: mcash/mapi_client/validation.py [hunk 4]
```diff
@@ -71,7 +71,20 @@ def wrapper(*args, **kwargs):
...
+        'description': basestring,
+        Required('vat_rate'): basestring,
```
Comment (mandus) [inline mcash/mapi_client/validation.py]: vat_rate is optional

File: mcash/mapi_client/validation.py [hunk 5]
```diff
@@ -71,7 +71,20 @@ def wrapper(*args, **kwargs):
...
+        Required('total'): basestring,
+        'tags': [{
+            Required('tag_id'): basestring,
```
Comment (mandus) [inline mcash/mapi_client/validation.py]: both tag_id and label are required

File: mcash/mapi_client/mapi_client.py (hunk 2 above)
Comment (aleksen) [inline mcash/mapi_client/mapi_client.py]: @mandus  Is the sum of the totals compared with the new amount, or the initial amount from the previous request?

Comment (mandus) [inline mcash/mapi_client/mapi_client.py]: You are not allowed to change the amount on a payment!

**Comment by mandus** [comment]
LGTM

**Pull request merged**